backend/inspirehep/snow/api.py

```diff
@@ -0,0 +1,571 @@
+        if not assignee:
+            self.edit_ticket(ticket_id, payload={"assigned_to": ""})
+
+    def _get_third_party_tickets_for_recid(self, recid):
+        """Returns all tickets that are associated with the given recid.
+
+        Args:
+            recid (int): The record id.
+        """
+        search_query = f"u_third_party_ticket_id={recid}"
+        ticket = self.search_snow(self.third_party_ticket_endpoint, search_query)
+        return ticket
+
+    def _prepare_query_string_from_parameters_dict(self, query_string_parameters):
+        """Helper to prepare request query string in a SNOW specific way
+        Args:
+            query_string_parameters (dict): query string parameters
+        Returns:
+            query_string (str): formatted query string
+        """
+        query_string = "^".join(
+            f"{key}={value}" for key, value in query_string_parameters.items()
+        )
+        return query_string
+
+    def get_tickets_by_recid(self, recid, exclude_resolved=True):
+        """Returns all tickets that are associated with the given recid.
+
+        Args:
+            recid (int): The record id.
+            exclude_resolved (boolean): should resolved tickets be excluded from the response
+        Returns:
+            tickets_for_recid (list): list of found tickets
+        """
+        third_party_search_query_string = f"u_third_party_ticket_id={recid}"
+        tickets_search_parameters = dict(u_third_party="INSPIRE")
+        if exclude_resolved:
+            tickets_search_parameters[
+                "u_current_task_state"
+            ] = f"{self.ticket_status_mapping['waiting']}^ORu_current_task_state={self.ticket_status_mapping['assigned']}"
+        try:
+            third_party_ticket = self.search_snow(
+                self.third_party_ticket_endpoint, third_party_search_query_string
+            )
+            if not third_party_ticket:
+                LOGGER.warning(
+                    "Third party ticket for recid was not found!", recid=recid
+                )
+                return []
+            tickets_search_parameters["u_third_party_ticket"] = third_party_ticket[0][
+                "sys_id"
+            ]
+            tickets_search_query = self._prepare_query_string_from_parameters_dict(
+                tickets_search_parameters
+            )
+            tickets_for_recid = self.search_snow(
+                self.ticket_endpoint, tickets_search_query
+            )
+            tickets_for_recid = [
+                self.get_ticket_simplified_response(ticket)
+                for ticket in tickets_for_recid
+            ]
+        except requests.exceptions.RequestException:
+            LOGGER.error("Can't find tickets for recid", recid=recid)
+            tickets_for_recid = []
+        return tickets_for_recid
+
+    def create_inspire_ticket(
+        self,
+        user_email=None,
+        functional_category=None,
+        subject=None,
+        description="",
```

Comment:
```suggestion
        description=None,
```

Reply:
this is on purpose actually, we use later `_strip_lines` on it which requires a string, if you prefer I can modify this function

Reply:
yes, let's move do a check in the `strip_lines`, it's better to have immutable defaults in the signature

Reply:
strings are immutable ;)

Reply:
you are absolute right, I had in mind array for some reason while replying 😅. Still let's move the logic inside the `strip_lines`